keps/sig-network/0006-20180314-coredns-GA-proposal.md

```diff
@@ -0,0 +1,134 @@
+* In case a user wants to install kube-dns instead of CoreDNS, they have to set the feature-gate of CoreDNS to false. `--feature-gates=CoreDNS=false`
+* For users upgrading Kubernetes via kubeadm, it will install CoreDNS whether the user had kube-dns or CoreDNS in a previous kubernetes version (Unless the CoreDNS feature-gate is disabled).
+* The configmap of a previously installed kube-dns will be automatically translated to the equivalent CoreDNS configmap.
+
+#### Kube-up
+
+* CoreDNS set as the default DNS service in place of kube-dns.
+
+#### Minikube
+
+* CoreDNS to be enabled by default in the add-on manager, with kube-dns disabled by default.
+ 
+### Implementation Details/Notes/Constraints 
+
+#### Use Cases
+
+* CoreDNS supports all functionality of kube-dns and also addresses several use-cases kube-dns lacks. Some of the Use Cases are as follows: 
+    * Making an alias for an external name [#39792](https://github.com/kubernetes/kubernetes/issues/39792)
+    * Dynamically adding services to another domain, without running another server [#55](https://github.com/kubernetes/dns/issues/55)
+    * Adding an arbitrary entry inside the cluster domain (for example TXT entries [#38](https://github.com/kubernetes/dns/issues/38))
+    * Verified pod DNS entries (ensure pod exists in specified namespace)
+    * Limit PTR replies to the cluster CIDR [#125](https://github.com/kubernetes/dns/issues/125)
+    * Serve DNS for selected namespaces [#132](https://github.com/kubernetes/dns/issues/132)
+    * Serve DNS based on a label selector.
+    * Support for wildcard queries (e.g., `*.namespace.svc.cluster.local` returns all services in `namespace`)
+    
+* By default, the user experience would be unchanged. For more advanced uses, existing users would need to modify the ConfigMap that contains the CoreDNS configuration file.
+* Since CoreDNS has more supporting features than kube-dns, there will be no path to retain the CoreDNS configuration in case a user wants to switch to kube-dns.
+
+#### Configuring CoreDNS
+
+The CoreDNS configuration file is called a `Corefile` and syntactically is the same as a [Caddyfile](https://caddyserver.com/docs/caddyfile). The file consists of multiple stanzas called _server blocks_.
+Each of these represents a set of zones for which that server block should respond, along with the list of plugins to apply to a given request. More details on this can be found in the 
+[Corefile Explained](https://coredns.io/2017/07/23/corefile-explained/) and [How Queries Are Processed](https://coredns.io/2017/06/08/how-queries-are-processed-in-coredns/) blog entries.
+
+### Unresolved questions
+
+* Fate of kube-dns in future releases, i.e. deprecation path.
+
+## Graduation Criteria
+
+* Passing all e2e conformance tests and DNS related.
+* Process in place for getting CoreDNS image in gcr.io.
+* CoreDNS be installed as the default DNS service.
```

Comment:
bullets 2 and 3 are goals...  do they belong in graduation criteria?  Seems redundant.